gbpservice/nfp/lifecycle_manager/modules/device_lifecycle_manager.py

```diff
@@ -0,0 +1,763 @@
+                                    nfd_request.get('share_existing_device'))
+        device_data['management_network_info'] = (
+                                    nfd_request.get('management_network_info'))
+
+        if network_function:
+            device_data['network_function_id'] = network_function['id']
+            device_data['service_chain_id'] = network_function['service_chain_id']
+
+        device_data['network_function_instance_id'] = (
+                                    network_function_instance['id'])
+        device_data['tenant_id'] = network_function_instance['tenant_id']
+
+        nsi_port_info = []
+        for port_id in network_function_instance.pop('port_info'):
+            port_info = self.nsf_db.get_port_info(self.db_session, port_id)
+            nsi_port_info.append(port_info)
+
+        device_data['ports'] = nsi_port_info
+
+        if service_vendor:
+            device_data['service_vendor'] = service_vendor
+        if service_type:
+            device_data['service_type'] = service_type
+        # TODO: Get these values from SLCM, it should be available in service
+        # profile.
+        device_data['compute_policy'] = 'nova'
+        # To get the network mode, fetch it from port_info
+        device_data['network_policy'] = nsi_port_info[0]['port_policy'].lower()
+
+        return device_data
+
+    def _get_nsf_db_resource(self, resource_name, resource_id):
+        db_method = getattr(self.nsf_db, 'get_' + resource_name)
+        return db_method(self.db_session, resource_id)
+
+    def _update_device_data(self, device, device_data):
+        device.update(device_data)
+        return device
+
+    # Create path
+    def create_network_function_device(self, event):
+        """ Returns device instance for a new service
+
+        This method either returns existing device which could be reused for a
+        new service or it creates new device instance
+        """
+        nfd_request = event.data
+        device = None
+
+        LOG.info(_("Received create network service device request with data"
+                   "%(data)s"), {'data': nfd_request})
+
+        device_data = self._get_device_data(nfd_request)
+        lifecycle_driver = self._get_lifecycle_driver(
+                                    device_data['service_vendor'])
+        dev_sharing_info = (
+            lifecycle_driver.get_network_function_device_sharing_info(
+                device_data))
+        if dev_sharing_info:
+            device = self._get_device_to_reuse(device_data, dev_sharing_info)
+            if device:
+                device = self._update_device_data(device, device_data)
+
+        # To handle case, when device sharing is supported but device not
+        # exists to share, so create a new device.
+        if dev_sharing_info and device:
+            # Device is already active, no need to change status
+            self._create_event(event_id='DEVICE_HEALTHY',
```

Comment:
Why do we post this event 'DEVICE_HEALTHY' instead of calling the handler?
Aren't methods posted for handlers across modules? If this is true, every create_event() should be checked for this and addressed.